kernel/src/hil/qdec.rs

```diff
@@ -0,0 +1,33 @@
+//! Interface for a Qdec compatible chip
+//!
+//! This trait provides a stanfard interface for chips with a
+//! quadrature decoder. Note this interface is experimental and
+//! may need further updates once implemented on additional chips
+
+use crate::returncode::ReturnCode;
+
+pub trait QdecDriver {
+    /// Sets the client which will receive interrupts
+    fn set_client(&self, client: &'static dyn QdecClient);
+
+    /// Enables the SAMPLERDY interrupt
+    fn enable_interrupts(&self) -> ReturnCode;
+
+    /// Enables the Qdec, returning error if Qdec does not exist
+    fn enable_qdec(&self) -> ReturnCode;
```

Comment:
Generally, we try to make hardware presence / failure a compile-time check rather than a runtime check in the kernel. If a kernel can access a QDEC object (as created by the board file), then one must physically exist (otherwise the board should not create it).